modules/bridge/src/main/ts/ephox/bridge/components/dialog/DialogFooterButton.ts

```diff
@@ -56,7 +63,14 @@ export interface DialogFooterMenuButton extends BaseDialogFooterButton {
   items: DialogFooterToggleMenuItem[];
 }
 
-export type DialogFooterButton = DialogFooterNormalButton | DialogFooterMenuButton;
+export interface DialogFooterTogglableIconButton extends Omit<BaseDialogFooterButton, 'icon'> {
+  type: 'togglableIconButton';
```

Comment:
Why not togglabelIconButton => togglableButton? since text, icon, and both are possible